notebooks/Manage Map Areas with Group and Index.ipynb

```diff
@@ -0,0 +1,456 @@
+    "\n",
+    "        if basemap_url is not None:\n",
+    "            # using the attribute values for min_scale and max_scale - loop through service supported levels and include\n",
+    "            \n",
+    "            try:\n",
+    "                basemap_levels_array = []\n",
+    "                map_service = MapImageLayer(basemap_url,gis)\n",
+    "                for lod in map_service.properties.tileInfo[\"lods\"]:\n",
+    "                    # min_scale and max_scale are required\n",
+    "                    if map_area.get_value(\"min_scale\") >= lod[\"scale\"] >= map_area.get_value(\"max_scale\"):\n",
+    "                        # include this level in basemap_levels\n",
+    "                        basemap_levels_array.append(str(lod[\"level\"]))\n",
+    "                        \n",
+    "            except Exception as e:  \n",
+    "                print(\"  failed - processing basemap levels\")\n",
+    "                print(e)\n",
+    "                \n",
+    "            basemap_levels = \",\".join(basemap_levels_array)\n",
+    "\n",
+    "            #set the web map tile service parameters url and levels\n",
+    "            map_area_tile_services.append({\"url\":basemap_url,\"levels\":basemap_levels})    \n",
+    "        \n",
+    "    return map_area_tile_services\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Processing each web map in group, as defined by index"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"-- begin processing map areas --\")\n",
+    "process_begin_time = datetime.datetime.now()       \n",
+    "\n",
+    "# optional - truncate the logs - this can be removed to based on retention preferences\n",
+    "#map_areas_hfl.tables[0].manager.truncate()     \n",
+    "\n",
+    "# get all group shared items\n",
+    "for item in manage_map_areas_group.content():\n",
+    "    # if the shared item is not a web map skip it \n",
+    "    if( item[\"type\"] != \"Web Map\" ):\n",
+    "        continue # to next item in for loop   \n",
+    "\n",
+    "    print(\"--------------\")\n",
+    "    print(f\"processing web map: {item['title']}\")        \n",
+    "\n",
+    "    # Cast item to Map\n",
+    "    web_map = Map(item)\n",
+    "\n",
+    "    # create map area as defined in map_areas_hfl\n",
+    "    where_clause =  f\"webmap_id IN ('{item.id}', 'ALL') and status = 'Active'\"\n",
+    "    map_area_layer = map_areas_hfl.layers[0]\n",
+    "    map_areas_feature_set = map_area_layer.query(where=where_clause) \n",
+    "    print(f\" map area count: {str(len(map_areas_feature_set.features))}\")\n",
+    "\n",
+    "    # check to ensure there are no more than 16 map areas, if there are skip to next map.\n",
+    "    if len(map_areas_feature_set.features) > 16:\n",
```

Comment:
Nice. Is there some how we could do a check for the 4GB limitation? Or that is already covered by the 16 count? Thank you!

Reply:
I'm not sure there is any way we can check before running the create map area process.  Maybe @justincolville  has an idea

Reply:
Unfortunately nope, no way to do this for mobile geodatabases/feature layers.  We could use estimate tiles for the vector and rater tile layers and see what the size would come out to.  You could look at that as a potential enhancement for the future maybe, but I wont worry about it right now.